File: src/julia/modules/Multistart/src/statistics.jl
function printStats(extmd::ExternalMSLPData)::Nothing
	@printf("Best solution value: %.2f\n", extmd.bestSol.value)
	@printf("Total time elapsed: %.6f\n", extmd.totalTimeElapsed)
	@printf("Time to best: %.6f\n", extmd.timeToBest)
	@printf("Iterations: %d\n", extmd.iteration)
	@printf("Iterations to best: %d\n", extmd.iterationToBest)
	@printf("Improvements: %d\n", extmd.countImprovements)
	@printf("Infeasible solutions percentage: (%d / %d) = %.4f %%\n", extmd.infeasibleSol, extmd.iteration, extmd.percentageInfeasibleSol)
	@printf("(LP improvements)/(LP runs) = %d/%d = %.4f %%\n", extmd.lpImpr, extmd.lpRuns, extmd.percentageLPImpr)
	@printf("Mean LP improvement percentage: %.4f %%\n", extmd.meanLPImprPercentage)

	return nothing
end

function calculateStats(extmd::ExternalMSLPData)::Nothing
	extmd.percentageInfeasibleSol = round(extmd.infeasibleSol / (extmd.iteration) * 100, digits = 4)
	extmd.percentageLPImpr = round(extmd.lpImpr / max(1, extmd.lpRuns) * 100, digits = 4)
	extmd.meanLPImprPercentage = round(extmd.sumLPImprPercentage / max(1, extmd.lpRuns) * 100, digits = 4)
	return nothing
end

File: src/python/modules/multistart/statistics.py
from .entities_mslp import ExternalMSLPData


def print_stats(extmd: ExternalMSLPData) -> None:
    print(f"Best solution value: {extmd.best_sol.value:.2f}")
    print(f"Total time elapsed: {extmd.totalTimeElapsed:.6f}")
    print(f"Time to best: {extmd.time_to_best:.6f}")
    print(f"Iterations: {extmd.iteration}")
    print(f"Iterations to best: {extmd.iterationToBest}")
    print(f"Infeasible solutions percentage: {extmd.percentageInfeasibleSol:.4f}")
    print(
        f"(LP improvements)/(LP runs) = {extmd.lpImpr}/{extmd.lpRuns} = {extmd.percentageLPImpr:.4f}"
    )
    print(f"Mean LP improvement percentage: {extmd.meanLPImprPercentage:.4f}")


def calculate_stats(extmd: "ExternalMSLPData") -> None:
    if extmd.iteration > 0:
        extmd.percentageInfeasibleSol = round(
            (extmd.infeasibleSol / extmd.iteration) * 100, 4
        )
    else:
        extmd.percentageInfeasibleSol = 0.0

    extmd.percentageLPImpr = round((extmd.lpImpr / max(1, extmd.lpRuns)) * 100, 4)
    extmd.meanLPImprPercentage = round(
        (extmd.sumLPImprPercentage / max(1, extmd.lpRuns)) * 100, 4
    )
